src/simulation/LoadGenerator.cpp

```diff
@@ -896,8 +943,11 @@ LoadGenerator::submitTx(GeneratedLoadConfig const& cfg,
                       tx->getInclusionFee());
             return false;
         }
+
+        // No re-submission in PAY_PREGENERATED mode
         if (++numTries >= TX_SUBMIT_MAX_TRIES ||
-            status != TransactionQueue::AddResultCode::ADD_STATUS_ERROR)
+            status != TransactionQueue::AddResultCode::ADD_STATUS_ERROR ||
+            cfg.mode == LoadGenMode::PAY_PREGENERATED)
```

Comment:
I think we need to guard the `maybeHandleFailedTx` and only call it if mode != PAY_PREGENERATED. It looks like that function makes unchecked dereferences to the `from` account pointer, which is always null for PAY_PREGENERATED.

Reply:
This `if` case already handles that by failing before we can get to `maybeHandleFailedTx`